motionplan/collision.go

```diff
@@ -312,3 +345,15 @@ func (cs *CollisionSystem) CollisionBetween(keyName, testName string) bool {
 	}
 	return false
 }
+
+// AddCollisionSpecification takes a Collision as an argument and either whitelists or blacklists collisions between the two entities.
+// Whether or not they are whitelisted or blacklisted depends on the value of the penetration depth of the given Collision.
+// If this value is positive the collision is whitelisted, otherwise it is blacklisted.
```

Comment:
Maybe blacklist/whitelist isn't the correct paradigm.  My thought is blacklist is "make sure that a collision between these entities is not something that we allow" and whitelist is "don't bother to check for collision because we're fine with it'. Does that make sense?

Reply:
I understand that; it's more that having that blacklist as a concept to be passed in doesn't do anything different/special, and doesn't really correspond to what we'll be passing in in the ConstraintSpecification API call.

I think for the purpose of this PR you can assume that something upstream consuming the API call creates the correct list of `*Collision` and passes them all here, and all can be treated as whitelists in this code.

Reply:
Just reread the scope doc and I was not understanding the intention behind the "deny" functionality.  

That being said, it feels weird to add a setter that only works in one direction.  So even if we are not going to be using the "blacklist" functionality, I feel that it is a useful feature.

Reply:
Ok I will remove